### ISSUE TITLE
for item in list in list

### ISSUE THREAD

**Issue by KommuSoft** [pylint-dev/pylint#10596]
### Current problem

In Python apparently one can write:

```
[v for v in list in list]
```

it groups `list in list`, so it is interpreted as:

```
[v for v in (list in list)]
```

but given I understand the specs of Python's data model correctly, it is mandatory that `item in collection` returns `True` or `False` (or raises a type error). So this does not make much sense, since a boolean is not iterable.

Should we add a detector for this? I stumbed upon this by looking through some code (I was actually suprised this was valid Python).

### Desired solution

add a detector for the pattern:
```
[item in list in list]
```

**Comment by Pierre-Sassoulas** [comment]
Right now pylint is raising 3 warnings on the example you gave:

a.py:
```python
numbers = [1, 2, 3]
a = [v for v in numbers in numbers]
```
Result :
```
a.py:2:12: E1133: Non-iterable value numbers in numbers is used in an iterating context (not-an-iterable)
a.py:2:0: R1721: Unnecessary use of a comprehension, use list(numbers in numbers) instead. (unnecessary-comprehension)
a.py:2:12: R0124: Redundant comparison - numbers in numbers (comparison-with-itself)
```

not-an-iterable seems to be doing what you want, no ?
